Sources/TensorFlow/Layers/Core.swift

```diff
@@ -166,6 +171,10 @@ public struct Dense<Scalar: TensorFlowFloatingPoint>: Layer {
     /// - Returns: The output.
     @differentiable
     public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
+        if weight.rank == 3 {
```

Comment:
@bgogul Is there a lazy tensor concern here?

Reply:
If believe there is, but what if I write this as: `weight.rankTensor == Tensor<Int>(3)`?

Reply:
Yeah, that might work a lot better, but I'll wait for @bgogul to confirm.

Reply:
Actually, given that `if` statements are not currently being lowered to TF ops, this should definitely affect lazy tensor because the condition depends on `weight`. I can move that condition to the initializer and add a boolean stored property named `batched`. Using that in the `if` statement will be safe with lazy tensor. @rxwei what do you think?

Reply:
Hmm, I'd very much prefer keeping the existing semantics if possible.

Reply:
This will retain the same semantics. It just moves the rank check to the initializer.

Reply:
Yes, this would cause materialization in lazy tensor, and thereby, break the trace. Even changing the condition to `weight.rankTensor == Tensor<Int>(3)` won't fix it. Adding the `batched` property would fix it though. 

That said, I don't think it is sustainable to have to rewrite code to help lazy tensor in the long run. If the code becomes too awkward and ugly, just do what makes the most sense. The right way to fix this would be to track shapes in the swift side.

Reply:
I just pushed the change so you can see what I mean.

Reply:
@bgogul Thanks for the confirmation! I already switched to the `batched` property. And yes, I agree with what you say. In this case I believe it's ok to modify the code given how prevalent dense layers are and also given that lowering if-statements to TF ops may take a while.